cpp/include/Ice/Communicator.h

```diff
@@ -340,23 +320,81 @@ namespace Ice
     private:
         static CommunicatorPtr create(InitializationData);
 
-        //
-        // Certain initialization tasks need to be completed after the
-        // constructor.
-        //
+        // Certain initialization tasks need to be completed after the constructor.
         void finishSetup(int&, const char*[]);
 
         [[nodiscard]] IceInternal::ReferencePtr _stringToProxy(std::string_view str) const;
         [[nodiscard]] IceInternal::ReferencePtr _propertyToProxy(std::string_view property) const;
 
+        /// @cond INTERNAL
         friend ICE_API CommunicatorPtr initialize(int&, const char*[], InitializationData);
-        friend ICE_API CommunicatorPtr initialize(StringSeq&, InitializationData);
         friend ICE_API CommunicatorPtr initialize(InitializationData);
+        /// @endcond
+
         friend ICE_API IceInternal::InstancePtr IceInternal::getInstance(const Ice::CommunicatorPtr&);
         friend ICE_API IceInternal::TimerPtr IceInternal::getInstanceTimer(const Ice::CommunicatorPtr&);
 
         const IceInternal::InstancePtr _instance;
     };
+
+    /// A helper class that uses Resource Acquisition Is Initialization (RAII) to hold a communicator instance, and
```

Comment:
Moved from Initialize.h. Didn't change except for doc-comments and a little more inlining.